File: [백준] 피보나치 함수(1003).cpp
#include <iostream>
#include <vector> 

using namespace std;

vector<pair<int,int>> fibos(41);

int main(void)
{
	int N, T;
	cin >> T;
	
	fibos[0] = make_pair(1,0);
	fibos[1] = make_pair(0,1);
	fibos[2] = make_pair(1,1);
	
	for(int i=3;i<42;++i) {
		fibos[i] = make_pair(fibos[i-1].first+fibos[i-2].first, fibos[i-1].second+fibos[i-2].second);	
	}
	
	cout << fibos[1].first << endl;
	for(int i=0;i<T;++i) {
		cin >> N;
		cout << fibos[N].first << ' ' << fibos[N].second << endl;
	}
	return 0;
}

File: [백준] 피보나치 함수(1003).py
fibos = [[] for _ in range(41)]
fibos[0] = [1,0]
fibos[1] = [0,1]
fibos[2] = [1,1]

for i in range(3,41):
    fibos[i] = [ fibos[i-1][0]+fibos[i-2][0], fibos[i-1][1]+fibos[i-2][1] ]

T = int(input())
for i in range(T):
    N = int(input())
    print(fibos[N][0], fibos[N][1])